sound/soc/sof/intel/hda-loader.c

```diff
@@ -184,6 +184,14 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, int stream_tag)
 		flags |= SOF_DBG_DUMP_FORCE_ERR_LEVEL;
 
 	hda_dsp_dump(sdev, flags);
+
```

Comment:
@plbossart I realize this is just moving the existing code to before the cores are powered off but  the call to hda_dsp_dump() that precedes this will already have printed the ext som status including the rom status and the rom error below. Should we still duplicate the log?

Reply:
you tell me @ranj063 ;-)

it's true that there is more information for the status, but I don't get why we read HDA_DSP_SRAM_REG_ROM_STATUS twice, once in hda_dsp_dump_ext_rom_status() and later in hda_get_get_status(). 

In addition I am puzzled by this 
````
	/* not for us, must be generic sof message */
	dev_dbg(sdev->dev, "unknown ROM status value %8.8x\n", status);
````
that doesn't quite right? We first log that message in hda_dsp_dump_ext_rom_status() but somehow can't find this value in a dictionary so silently drop it. Wondering if the dictionary is complete then? What was the point of logging a value but not that it's unknown?

And going back to the duplication, we don't read HDA_DSP_SRAM_REG_ROM_ERROR anywhere else, which is quite puzzling to me as well. Shouldn't the error be added to hda_dsp_dump_ext_rom_status()?

I wonder if we are reading the wrong registers to be honest.

Reply:
@plbossart maybe we need a clean up of how and what we dump in case of errors. 

We used to have the unknown ROM status message before we fixed our IPC handler, not sure if we still have that problem. I have not seen any recent reports of this lately. 

But you're right, there is no need to call hda_dsp_get_status() twice. We should print the decoded ROM status , followed by the extended ROM status which would also contain the the  ROM_ERROR. Do you want me to clean this up?